### PR TITLE
Fix division-by-zero on import

### LINKED ISSUE #4777: Division by zero during Pulp import
**Version**
3.36.0+

**Describe the bug**

During a Katello test run, the following exception was encountered

```
 {"traceback"=>"  File \"/usr/lib/python3.11/site-packages/pulpcore/tasking/tasks.py\", line 61, in _execute_task\n    result = func(*args, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/site-packages/pulpcore/app/tasks/importer.py\", line 453, in pulp_import\n    with tarfile.open(path, \"r\", fileobj=fp) as tar:\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib64/python3.11/tarfile.py\", line 1815, in open\n    fileobj.seek(saved_pos)\n  File \"/usr/lib/python3.11/site-packages/pulpcore/app/tasks/importer.py\", line 132, in seek\n    self.chunk = target // self.chunk_size\n                 ~~~~~~~^^~~~~~~~~~~~~~~~~\n", "description"=>"integer division or modulo by zero"} (Katello::Errors::Pulp3Error)
```

I'm not 100% certain, but I suspect the cause here is that if chunks aren't used during the export, chunk_size is set to 0:

https://github.com/pulp/pulpcore/blob/main/pulpcore/app/tasks/export.py#L456C31-L456C31

and ChunkedFile reads that value here:

https://github.com/pulp/pulpcore/blob/main/pulpcore/app/tasks/importer.py#L79

and we're using ChunkedFile even in the non-chunked case so long as a TOC file was provided.

https://github.com/pulp/pulpcore/blob/main/pulpcore/app/tasks/importer.py#L335-L336

**To Reproduce**

I would expect it to be reproducible if performing a non-chunked export, and providing a TOC path for the import.

**Expected behavior**

We should never see an integer division by zero error.

**Additional context**

### PR DESCRIPTION
**Pull request by dralley**
closes #4777

TODO: still working on the tests, it's a bit difficult to piece together how the fixtures interact...

### CODE REVIEW

File: pulpcore/app/tasks/export.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ def pulp_export(exporter_pk, params):
             if the_export.validated_chunk_size:
                 chunk_size = the_export.validated_chunk_size
             else:
-                chunk_size = 0
+                chunk_size = os.path.getsize(tarfile_fp)
```
Comment (mdellweg) [inline pulpcore/app/tasks/export.py:456]: This is on the export side. Do we want to account for this on the import side too, so we can handle existing exports?

Comment (dralley) [reply pulpcore/app/tasks/export.py:456]: Existing imports aren't a huge concern, but I'm definitely not against defense in depth.  But which approach to use?

We could set it null and have the import side figure it out.  Or we could have the import side not use ChunkedFile when there's no chunks, or 1 chunk.  Or we could have the ChunkedFile ignore the chunking logic in some circumstances.  

Separately, we could also make `chunk_size` required if you make a chunked export, or we could set a default value.  Currently if there is no chunk size provided, there will just be one chunk.

Comment (mdellweg) [reply pulpcore/app/tasks/export.py:456]: I'd be fine with any of these solutions.

Comment (ggainey) [reply pulpcore/app/tasks/export.py:456]: > I'd be fine with any of these solutions.

My $0.02 would be to do this same thing on the import-side - if chunbked_size==0 set it to filesize. That addresses the issue, and lets the rest of the codeflow work without needing special-casing.

**Comment by dralley** [comment]
After some discussion with the Katello folks, the most convenient thing to do at the moment is to merge this without tests and follow up with those tests in a separate PR.  So that's what I'll do.

File: pulpcore/app/tasks/export.py [hunk 2]
```diff
@@ -450,23 +448,20 @@ def pulp_export(exporter_pk, params):
...
-            if the_export.validated_chunk_size:
-                chunk_size = the_export.validated_chunk_size
-            else:
-                chunk_size = 0
-            chunk_toc = {
+            table_of_contents = {
                 "meta": {
-                    "chunk_size": chunk_size,
-                    "file": os.path.basename(tarfile_fp),
-                    "global_hash": tarfile_hash,
```
Comment (dralley) [inline pulpcore/app/tasks/export.py:461]: `global_hash` and `file` are no longer used in the import path anywhere

File: pulpcore/app/tasks/export.py [hunk 3]
```diff
@@ -450,23 +448,20 @@ def pulp_export(exporter_pk, params):
...
                     "checksum_type": checksum_type,
                 },
                 "files": {},
             }
+
+            if the_export.validated_chunk_size:
+                table_of_contents["meta"]["chunk_size"] = the_export.validated_chunk_size
```
Comment (dralley) [inline pulpcore/app/tasks/export.py:459]: Set `chunk_size` only if it was provided

**Review by mdellweg — APPROVED** [review]

**Review by ggainey — APPROVED** [review]

**Pull request merged**